### ISSUE TITLE
Add the same domain to multiple designated containers

### ISSUE THREAD

**Issue by lieryan** [mozilla/multi-account-containers#1887]
The "Limit to Designated Sites" option is a great option, but enabling this causes problems if you have multiple containers that shares the sa
me Single Sign On (SSO) domain.

Currently, I want to have Container A to only ever run Site A1, A2, A3, and Container B to only ever run Site B1, B2, B3, so I have "Limit to Designated Sites" option turned on for these containers. However, Site A1 and B1 needs to occasionally redirect to Site S, for example, to do a SSO login. Currently, you can only assign a single domain to a single container, which means that if you add Site S to Container A, then add it again to Container B, then Site A1 can no longer redirect for login correctly to Site C.

**Comment by Scorbie** [comment]
Seconded as I use multiple containers for the same site (with different accounts) and this feature would be very helpful for me.

**Comment by wfaulk-bw** [comment]
Can you not just leave Site S unassigned? That way when Site A redirects there, it stays in Container A, and when Site B redirects there, it stays in Container B.

**Comment by tmottabr** [comment]
> 
> 
> Can you not just leave Site S unassigned? That way when Site A redirects there, it stays in Container A, and when Site B redirects there, it stays in Container B.

not if in one of the containers you want to enable "limit to designed sites" as well so links to external sites do not open in that same container..

If you do that it cannot open the site in that container at all if the site is not assigned to that container.

And if you assign it to that container you cannot open it in any another container as it get locked to that container it is assigned to.

basically what we need is either decouple the configuration of what sites will be forced to open in a given container to the list of sites that a container should be limited to.

Or allow the same site to be linked to two separated containers.

my problem is exactly this, i have a site i have two accounts but i want to have the two containers i use for this site configured with "limited to designated sites", but currently i cannot because i can only link the domain to one container and if i do i cannot open it in the other.

**Comment by tmottabr** [comment]
some people also mentioned the SSO problem, where your authentication is performed by an third-party.

You might have two containers that you want to limit to designated sites but both sites authenticate with the same SSO provider.

If enable designated sites you cannot authenticate in the container that have it enabled and if you assign the SSO provider to one container you cannot authenticate in the other.

**Comment by ncallaway** [comment]
I think this is essentially the same request as #1892.

I've started working on a PR for this feature, and have a working proof of concept. I need to do a bit of cleaning up, so I expect to have a PR out by next weekend.

**Comment by ncallaway** [comment]
Draft PR is out here: https://github.com/mozilla/multi-account-containers/pull/2033. Remaining work is just the new icon and any PR feedback, which I'll work on this week.

**Comment by ncallaway** [comment]
The related PR is now ready for review.

**Comment by dannycolin** [comment]
Duplicate of #1749
